apps/web/pages/event-types/index.tsx

```diff
@@ -675,7 +679,11 @@ const EventTypeListHeading = ({
       <Avatar
         alt={profile?.name || ""}
         href={teamId ? `/settings/teams/${teamId}/profile` : "/settings/my-account/profile"}
-        imageSrc={`${WEBAPP_URL}/${profile.slug}/avatar.png` || undefined}
+        imageSrc={
+          loggedInUser?.username === profile.slug
```

Comment:
I don't understand the reasoning behind this change. `${WEBAPP_URL}/${profile.slug}/avatar.png` is supposed to serve the correct avatar. Why is it not working?

Reply:
@hariombalhara Yes `${WEBAPP_URL}/${profile.slug}/avatar.png` should have returned the correct avatar. It does return the correct avatar till the time user doesn't change the avatar in the current session. Due to session caching in the browser, it's not updating the image that the logged-in user has changed in the profile settings of a currently active session.

Reply:
@Keyurkd is there any way we can invalidate the avatar in the session cache when user updates it

Reply:
I don't think it is about the session. The problem is that the endpoint avatar.png is actually a [dynamic endpoint](https://github.com/calcom/cal.com/blob/5362cd66b4ab23392f1ad5ce5d5eca84aad0ab2a/apps/web/pages/api/user/avatar.ts#L68) that decides what's the profile pic for a user.  It currently has no cache-control headers and thus browsers are free to cache it. 

Making it completely non-cacheable will fix this issue but it would unnecessarily make function invocations for something that rarely changes.

This needs an RFC, but one quicker approach in our context can be that change the URL of the image every time the avatar changes. e.g. we compute and md5 of the image being uploaded and use that as query param in front of URL. Now, even if it is fully cached, changing the query param would be considered a different resources in terms of cache and will be fetched and cached again. We would have to store that md5 per user.

Also, if the issue is about the session not updating as well then there is a an update method available that might help https://next-auth.js.org/getting-started/client#updating-the-session